sdk/core/core-client/src/deserializationPolicy.ts

```diff
@@ -259,15 +259,18 @@ function handleErrorResponse(
   });
 
   // If the item failed but there's no error spec or default spec to deserialize the error,
+  // and the parsed body doesn't look like an error object,
   // we should fail so we just throw the parsed response
-  if (!errorResponseSpec) {
+  if (
+    !errorResponseSpec &&
+    !(parsedResponse.parsedBody?.error?.code && parsedResponse.parsedBody?.error?.message)
+  ) {
     throw error;
   }
 
-  const defaultBodyMapper = errorResponseSpec.bodyMapper;
-  const defaultHeadersMapper = errorResponseSpec.headersMapper;
-
   try {
+    const defaultBodyMapper = errorResponseSpec?.bodyMapper;
+    const defaultHeadersMapper = errorResponseSpec?.headersMapper;
```

Comment:
nitpick: is there a reason to retry these lines? seems they never throw.

Reply:
I moved them for some changes that were reverted, but I forgot to move them back out of the try-block.

Reply:
restored to old place.